### ISSUE TITLE
Incorrect use of center in child elements in initializer

### ISSUE THREAD

**Issue by skjolber** [buildmobile/iosrangeslider#1]
The center of an element is relative to its siblings, not its children; adding the slider to anywhere except y = 0 yields incorrect positions of subviews. So rather change initializer to this: 
- (id)initWithFrame:(CGRect)frame
  {
  self = [super initWithFrame:frame];
  if (self) {
      // Set the initial state
      _minThumbOn = false;
      _maxThumbOn = false;
      _padding = 20;
  
  ```
  _trackBackground = [[[UIImageView alloc] initWithImage:[UIImage imageNamed:@"bar-background.png"]] autorelease];
  _trackBackground.frame = CGRectMake((frame.size.width - _trackBackground.frame.size.width) / 2, (frame.size.height - _trackBackground.frame.size.height) / 2, _trackBackground.frame.size.width, _trackBackground.frame.size.height);
  [self addSubview:_trackBackground];
  
  _track = [[[UIImageView alloc] initWithImage:[UIImage imageNamed:@"bar-highlight.png"]] autorelease];
  _track.frame = CGRectMake((frame.size.width - _track.frame.size.width) / 2, (frame.size.height - _track.frame.size.height) / 2, _track.frame.size.width, _track.frame.size.height);
  [self addSubview:_track];
  
  _minThumb = [[[UIImageView alloc] initWithImage:[UIImage imageNamed:@"handle.png"] highlightedImage:[UIImage imageNamed:@"handle-hover.png"]] autorelease];
  _minThumb.frame = CGRectMake(0,0, self.frame.size.height,self.frame.size.height);
  _minThumb.contentMode = UIViewContentModeCenter;
  _minThumb.center = CGPointMake([self xForValue:selectedMinimumValue], self.frame.size.height / 2);
  [self addSubview:_minThumb];
  
  _maxThumb = [[[UIImageView alloc] initWithImage:[UIImage imageNamed:@"handle.png"] highlightedImage:[UIImage imageNamed:@"handle-hover.png"]] autorelease];
  _maxThumb.frame = CGRectMake(0,0, self.frame.size.height,self.frame.size.height);
  _maxThumb.contentMode = UIViewContentModeCenter;
  _maxThumb.center = CGPointMake([self xForValue:selectedMaximumValue], self.frame.size.height / 2);
  [self addSubview:_maxThumb];
  
  [self updateTrackHighlight];
  ```
  
  }
  return self;
  }

it seems to be working.

**Comment by snikch** [comment]
Hey, I've tried adding the slider to an arbitrary frame size, and position, and it still renders correctly for me. Can you show me some code that shows it in an incorrect place please?

**Comment by skjolber** [comment]
Yeah at first glance, it seems to work, however try to set the background of the view to for example red - you will see that the image views are not within the parent view. When I used a fixed height, I observed at 2x sensitivity in the y position; 1x for the parent view and 1x for the child elements.  Try adding it to the bottom of the screen; it will not be visible.

```
    [rangeSlider initWithFrame:CGRectMake(0, self.view.frame.size.height - 30.0, self.view.frame.size.width, 30.0)];
```

Using child.center = parent.center is not logical, although it really 'sounds' logical ;)
